modules/library/metadata/src/main/java/org/geotools/util/SoftValueHashMap.java

```diff
@@ -660,9 +602,8 @@ public void remove() {
          * or is about to be removed.</li>
          */
         final Object getAndClear() {
-            assert Thread.holdsLock(hash);
             final Object value = get();
-            super.clear();
+            clear();
```

Comment:
See my comment above, it's likely that the original code was correct, cleaning the reference without running the cleaner.